pkg/oc/bootstrap/clusterup/componentinstall/context.go

```diff
@@ -0,0 +1,98 @@
+package componentinstall
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	kclientcmd "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+	aggregatorclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	kinternal "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	"github.com/openshift/origin/pkg/oc/errors"
+	securityclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+)
+
+type Context interface {
+	GetKubeAdminClient() (kubernetes.Interface, error)
+	GetInternalKubeAdminClient() (kinternal.Interface, error)
+	GetAggregatorClient() (aggregatorclient.Interface, error)
+
+	AddPrivilegedUser(namespace, name string) error
+
+	MasterConfigDir() string
```

Comment:
don't mind... it will mean every component will have to do `masterConfigDir := path.Join(c.InstallContext.BaseDir(), kubeapiserver.KubeAPIServerDirName)`